### PR TITLE
OAUTH2-242 Use new util intead of PortalCache

### PR DESCRIPTION
**Pull request by csierra**

### CODE REVIEW

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/CodeGrantsClusterUtil.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+			return _getCodeGrant(code);
+		}
+
+		Future<ServerAuthorizationCodeGrant> future =
+			ClusterMasterExecutorUtil.executeOnMaster(
+				new MethodHandler(_getCodeGrantMethodKey, code));
+
+		try {
+			return future.get(1, TimeUnit.MINUTES);
```
Comment (topolik) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/CodeGrantsClusterUtil.java]: https://stackoverflow.com/questions/13582409/http-client-timeout-and-server-timeout: The default request timeout for Firefox is 30s (network.http.pipelining.read-timeout).

Maybe we should we make it configurable with default timeout 30 secs?

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/CodeGrantsClusterUtil.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+			}
+		}
+
+		return serverAuthorizationCodeGrants;
+	}
+
+	private static void _putCodeGrant(
+		ServerAuthorizationCodeGrant serverAuthorizationCodeGrant) {
+
+		_cleanup();
```
Comment (topolik) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/CodeGrantsClusterUtil.java]: Unnecessary. Just saying, we can leave it here.

**Comment by topolik** [comment]
Code style one: Do we need to hop from the public method right into private one? We usually go public -> protected.

**Comment by topolik** [comment]
ci:test:sf

**Comment by topolik** [comment]
ci:test:relevant

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/CodeGrantsClusterUtil.java (hunk 2 above)
Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/CodeGrantsClusterUtil.java]: what do you mean?

Comment (topolik) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/CodeGrantsClusterUtil.java]: nvm, just that we don't need to clean the queue on put.

Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/CodeGrantsClusterUtil.java]: ah... I see...

**Comment by topolik** [comment]
Pull request submitted at: https://github.com/brianchandotcom/liferay-portal/pull/69260
View total diff: https://github.com/topolik/liferay-portal/compare/19bbbf1ad1...2fdf3daf61

**Pull request closed without merging**